src/e2e/java/teammates/e2e/util/JMeterConfig.java

```diff
@@ -137,11 +148,58 @@ public HashTree createTestPlan() {
         // Create Test plan
         HashTree testPlanHashTree = new ListedHashTree();
         HashTree threadGroupHashTree = testPlanHashTree.add(testPlan, threadGroup);
+
+        // Add Config elements
         threadGroupHashTree.add(csvDataSet);
         threadGroupHashTree.add(cookieManager);
         threadGroupHashTree.add(defaultSampler);
-        threadGroupHashTree.add(onceOnlyController, loginSampler);
-        threadGroupHashTree.add(apiSampler);
+
+        // Add HTTP samplers (and configurations, pre/post processors)
+        HashTree loginSamplerHashTree = threadGroupHashTree.add(onceOnlyController, loginSampler);
+        HashTree apiSamplerHashTree = threadGroupHashTree.add(apiSampler);
+
+        switch (getTestMethod()) {
+        case HttpDelete.METHOD_NAME:
+            // fallthrough
+        case HttpPut.METHOD_NAME:
+            // fallthrough
+        case HttpPost.METHOD_NAME:
+            // Regex Extractor for CSRF token
+            RegexExtractor regexExtractor = new RegexExtractor();
+            regexExtractor.setName("Regular Expression Extractor");
+            regexExtractor.setUseField("true"); // Find matches in response headers
+            regexExtractor.setRefName("csrfToken");
+            regexExtractor.setRegex("Set-Cookie: CSRF-TOKEN=(.+?);");
+            regexExtractor.setTemplate("$1$");
+            regexExtractor.setProperty(TestElement.TEST_CLASS, RegexExtractor.class.getName());
+            regexExtractor.setProperty(TestElement.GUI_CLASS, RegexExtractorGui.class.getName());
+
+            // HTTP Header Manager
+            HeaderManager headerManager = new HeaderManager();
+            headerManager.setName("HTTP Header Manager");
+            headerManager.add(new Header("X-CSRF-TOKEN", "${csrfToken}"));
+            headerManager.setProperty(TestElement.TEST_CLASS, HeaderManager.class.getName());
+            headerManager.setProperty(TestElement.GUI_CLASS, HeaderPanel.class.getName());
+
+            String requestBody = getTestEndpointRequestBody();
+            if (requestBody == null || requestBody.isEmpty()) {
+                // Add Request Body
+                apiSampler.addNonEncodedArgument("", requestBody, "");
+                apiSampler.setPostBodyRaw(true);
+
+                // Add corresponding Request Header
+                // TODO: Parameterise content type when refactoring elements into methods
+                headerManager.add(new Header("Content-Type", "text/plain"));
```

Comment:
The `Content-Type` should in fact be defaulted to `application/json` as most of our API endpoints accept JSON. Only the student enrollment endpoint accepts `text/csv` (not `text/plain`) request body.

Reply:
> How exactly?

Currently, a "default structure" (shown below) is generated for each L&P test, with some configurable values (like the endpoint, number of threads). This works fine now since it's a single HTTP request for viewing profile, or enrolling students.

<img width="200" alt="2019-04-10_00h30_00" src="https://user-images.githubusercontent.com/29021586/55817770-cb0d2800-5b27-11e9-880e-d53bab432bdd.png">

---

Based on some investigation that we did yesterday, for viewing/submitting sessions, there will be multiple HTTP requests. Which is why we'll need to add more `HTTPSamplerProxy` elements (like `apiSampler`) to the test plan HashTree. Hence, the need to modularise the creation of elements into methods.

<img width="488" alt="2019-04-10_00h32_11" src="https://user-images.githubusercontent.com/29021586/55817924-17586800-5b28-11e9-806f-5cd4c99b8ac3.png">

_Multiple http requests when submitting feedback session_

<img width="250" alt="2019-04-10_00h35_27" src="https://user-images.githubusercontent.com/29021586/55818358-ffcdaf00-5b28-11e9-9bbe-e2053ce48158.png">

_What the corresponding test should be (most likely)_

Reply:
> It does not warrant being set manually for different endpoints.

👍 

> What do you mean "actual csrfToken key"?

Bad wording... I meant the the `csrf token` UUID

What I meant to say was that the header `X-CSRF-TOKEN` can be empty to successfully run the test. There is no need to use the regex and add each thread's `csrf token` to the request header.

Both of these work, but the first method (no token) seems hacky, and it isn't the right way of doing it.

![image](https://user-images.githubusercontent.com/29021586/55818833-24765680-5b2a-11e9-8e5d-854599355cd2.png)


![image](https://user-images.githubusercontent.com/29021586/55818752-eda04080-5b29-11e9-9132-4929337a3acf.png)

Reply:
> Both of these work, but the first method (no token) seems hacky, and it isn't the right way of doing it.

The fact that you can do the first method is a problem.